provider/provider.go

```diff
@@ -554,3 +542,16 @@ func hashContext(dockerContextPath string, dockerfile string) (string, error) {
 	// create a hash of the entire input of the hash accumulator
 	return accumulator.hexSumContext(), nil
 }
+
+func getIgnore(dockerIgnorePath string) ([]string, error) {
+	var ignorePatterns []string
+	dockerIgnore, err := os.ReadFile(dockerIgnorePath)
+	if err != nil && !os.IsNotExist(err) {
+		return ignorePatterns, fmt.Errorf("unable to read %s file: %w", dockerIgnorePath, err)
+	}
```

Comment:
nit: on my first read through this function, it looked like we'd error if `os.IsNotExist`. Is this more readable?

```suggestion
	if err != nil {
	  if os.IsNotExist(err) {
	  	// .dockerignore files are optional
			return "", nil
		}
		return ignorePatterns, fmt.Errorf("unable to read %s file: %w", dockerIgnorePath, err)
	}
```

Reply:
I'm here for more readability so yes, thank you! (gotta be a []string tho)